common/libs/signing.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import EthTx from 'ethereumjs-tx';
+import { sha3, ecsign } from 'ethereumjs-util';
+import { isValidRawTx } from 'libs/validators';
+
+export function signRawTxWithPrivKey(privKey: Buffer, rawTx: Object): string {
+  if (!isValidRawTx(rawTx)) {
+    throw new Error('Invalid raw transaction');
+  }
+
+  let eTx = new EthTx(rawTx);
```

Comment:
V3 Implementation for reference: https://github.com/kvhnuke/etherwallet/blob/3ab97999a157b1e0410de49b40080bcd35a1f5b3/app/scripts/uiFuncs.js#L113